swap/swap.go

```diff
@@ -542,7 +546,18 @@ func (s *Swap) getContractOwner(ctx context.Context, address common.Address) (co
 	return contr.Issuer(nil)
 }
 
-// Deploy deploys the Swap contract
+// NewInstanceAt creates a new instance of an already existing chequebook contract at address and sets chequebookAddr
+func (s *Swap) NewInstanceAt(address common.Address, backend swap.Backend) error {
```

Comment:
i am not going to ask you to change the name here (you know my position on this :stuck_out_tongue: ) but i _will_ say that i find it a bit weird that we have a function named `NewInstanceAt` which itself calls _another_ function called `InstanceAt`.

Reply:
Changed it to be just `InstanceAt`